sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -157,4 +157,42 @@ class PlannerSuite extends SparkFunSuite {
     val planned = planner.TakeOrderedAndProject(query)
     assert(planned.head.isInstanceOf[execution.TakeOrderedAndProject])
   }
+
+  test("PartitioningCollection") {
+    // First, we disable broadcast join.
+    val origThreshold = conf.autoBroadcastJoinThreshold
```

Comment:
This test could use the new SQLTestUtils `withConf` and `withTempTable` helper functions, I think.